### PR TITLE
fix disenchant filters for non-disenchantable items

### PR DESCRIPTION
**Pull request by Artur91425**
Sorry, the previous PR itself closed after renaming the branch ...

### CODE REVIEW

File: util/filter.lua [hunk 1]
```diff
@@ -154,7 +154,7 @@ M.filters = {
         validator = function(amount)
             return function(auction_record)
                 local item_info = info.item(auction_record.item_id)
-                local disenchant_value = item_info and disenchant.value(item_info.item_id, item_info.slot, item_info.quality, item_info.level)
+                local disenchant_value = item_info and disenchant.value(auction_record.item_id, item_info.slot, item_info.quality, item_info.level)
```
Comment (shirsig) [inline util/filter.lua:157]: Why was this necessary, shouldn't it be the same id?

File: core/disenchant.lua [hunk 2]
```diff
@@ -81,7 +81,7 @@ function M.value(item_id, slot, quality, level)
 end
 
 function M.distribution(item_id, slot, quality, level)
-    if IGNORE[item_id] or not ARMOR[slot] and not WEAPON[slot] or level == 0 then
+    if not item_id or IGNORE[item_id] or not ARMOR[slot] and not WEAPON[slot] or level == 0 then
```
Comment (shirsig) [inline core/disenchant.lua:84]: Why would it ever be called with an undefined item id? I think it should be fixed there rather than here.

File: util/filter.lua (hunk 1 above)
Comment (Artur91425) [reply util/filter.lua:157]: Because the `Item_info` table does not have the `item_id` field. Therefore, in `disenchant.value`, the first argument has always had the value of `nil`.

File: core/disenchant.lua (hunk 2 above)
Comment (Artur91425) [reply core/disenchant.lua:84]: This check is additional protection against improper functioning. If something in the addon will work incorrectly and the first argument will be `nil`, then this check will prevent the code execution below for the case when do not need to do. It is precisely because of the lack of such a check, I immediately did not find this problem that I use an incorrect table in which there is no `item_id` field

Comment (shirsig) [reply core/disenchant.lua:84]: I see, but I would prefer an assertion then like assert(item_id, "item_id parameter is mandatory")

File: util/filter.lua (hunk 1 above)
Comment (shirsig) [reply util/filter.lua:157]: Ah yes, you're right, I thought it did.

File: core/disenchant.lua (hunk 2 above)
Comment (Artur91425) [reply core/disenchant.lua]: made)

**Comment by shirsig** [comment]
Thanks :)
